### PR TITLE
Expand description of data_checksums GUC with checksum behavior and scope

### PR DESCRIPTION
**Pull request by litzell**
Addresses additional comments on closed PR: https://github.com/greenplum-db/gpdb/pull/2933#discussion_r132796314

### CODE REVIEW

**Comment by Eulerizeit** [comment]
Assuming the ignore GUC isn't set it should prevent mirroring.

-- Rob

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 1]
```diff
@@ -1299,7 +1299,24 @@
   <topic id="data_checksums">
     <title>data_checksums</title>
     <body>
-      <p>Reports whether heap checksums are enabled in the database system. </p>
+      <p>Reports whether checksums are enabled for heap data storage in the database system.
+        Checksums for heap data are enabled or disabled when the database system is initialized and
+        cannot be changed. When enabled, Greenplum Database computes and stores checksums for heap
+        data blocks on the Greenplum Database master and standby instances, and on all primary and
+        mirror segments. </p>
```
Comment (asimrp) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: Correction - checksums are computed only on primary, mirror simply writes the block sent by primary.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 2]
```diff
@@ -1299,7 +1299,24 @@
...
+      <p>When a block is later read from storage, the checksum is verified. Unless the
+          <codeph>ignore_checksum_failure</codeph> configuration parameter has been set to on, a
+        failed checksum verification generates an error and terminates the transaction, ensuring
+        that the corrupted data is not propagated to a standby or mirror segment. </p>
```
Comment (asimrp) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: Clarification - what prevents propagation of corruption is the corrupted page is not allowed to enter PostgreSQL managed memory.  No transaction can read it or write to it.  With `ignore_checksum_failure = on`, a page with invalid checksum is allowed to enter PostgreSQL memory.  The first update to this page will cause the page to be mirrored and therefore, propagation of corruption to mirror.

Comment (litzell) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: @asimrp The description for ignore_checksum_failure contains the statement "If the page header is corrupt an error is reported even if this option is enabled." This comes from the postgresql doc. If the header is corrupt, will ignore_checksum_failure=on still allow the page to be read into mem?

**Pull request merged**